packages/lodestar-beacon-state-transition/src/fast/index.ts

```diff
@@ -19,13 +19,20 @@ export function fastStateTransition(
   verifySignatures = true,
 ): IStateContext {
   const types = epochCtx.config.types;
-  const block = signedBlock.message;
+
+  // convert to tree-backed block: the root is computed multiple times,
+  // the contents are hash-tree-rooted multiple times, and some of the contents end up in the state as tree-form.
+  const {SignedBeaconBlock} = types;
+  const signedBlockCached = SignedBeaconBlock.tree.asTreeBacked(SignedBeaconBlock.tree.fromStructural(signedBlock));
```

Comment:
We should probably do this in validate step of blockprocessing as we calculate root there few times already and state transition won't care about backing then

Reply:
And I am doing it explicitly for the eth1 vote of the block, to make hash tree root cached, and do many repeated equality checks for the vote counting.
If the whole block ends up being tree backed anyway, the eth1 data change can be simplified.